File: 0896-monotonic-array/0896-monotonic-array.rs
impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
        let mut incr : i32 = 0;
        let mut decr : i32 = 0;
        for i in 0..nums.len()-1 {
            if nums[i] > nums[i+1] {
                decr+=1;
            }
            else if nums[i] < nums[i+1] {
                incr+=1;
            }
        }
        // println!("{incr} {decr}");
        if incr !=0 && decr != 0 {
            return false;
        }
        return true;
    }
}

File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        vector<int> nums1 = nums;
        sort(nums1.begin(),nums1.end());
        vector<int> nums2 = nums;
        sort(nums2.begin(),nums2.end(),greater<int>());
        return nums==nums1 || nums==nums2;
    }
};